### DISCUSSION TITLE
Unable to log in on new devices

### DISCUSSION THREAD

**Discussion by xeals** [dani-garcia/vaultwarden#3113] [Q&A] [answered]
I'm attempting to set up a new device and found that:

- I cannot log in at all: on the Android app, mobile browser, and desktop browser and extension.
- Attempting to log in will prompt for my password and 2FA, hang for a minute, and time out.
- Existing sessions are completely intact and can access my vault, export, etc..
- The admin page still works.

The issue appears similar to #2630 but 

I have tried:

- Using multiple authentication methods available to me:
  - Yubikey;
  - TOTP;
  - Newly configured TOTP;
  - No 2FA.
- Investigating debug/trace logs.
  - Nothing of interest appears, but happy to share them.
- Using multiple versions of the Android app (excluding the latest, as it requires server v1.27.0, which is currently unavailable to me).

Given the circumstances, I'm not willing at this stage to attempt anything that will invalidate existing sessions, such as changing my password.

Possible context:

- I have, in the last few months, migrated this instance from a different server in-place. I don't recall any issues immediately after the migration but I'm also not sure I've attempted a clean login since.
- The nginx configuration is equivalent to the Caddy configuration on the old server, and is identical to the configurations on the deployment wiki.

Symptomatic logs:

```
Jan 06 15:37:50 vaultwarden[1678133]: [2023-01-06 15:37:50.146][request][INFO] POST /api/accounts/prelogin
Jan 06 15:37:50 vaultwarden[1678133]: [2023-01-06 15:37:50.146][response][INFO] (prelogin) POST /api/accounts/prelogin => 200 OK
Jan 06 15:37:50 vaultwarden[1678133]: [2023-01-06 15:37:50.711][request][INFO] POST /identity/connect/token
Jan 06 15:37:50 vaultwarden[1678133]: [2023-01-06 15:37:50.750][error][ERROR] 2FA token not provided
Jan 06 15:37:50 vaultwarden[1678133]: [2023-01-06 15:37:50.750][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
Jan 06 15:37:54 vaultwarden[1678133]: [2023-01-06 15:37:54.472][request][INFO] POST /identity/connect/token
Jan 06 15:38:54 nginx[1679545]: 2023/01/06 15:38:54 [error] 1679545#1679545: *20 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 120.17.109.3, server: bitwarden.***, request: "POST /identity/connect/token HTTP/1.1", upstream: "http://127.0.0.1:41729/identity/connect/token", host: "bitwarden.***"
```

I am running Vaultwarden on NixOS with the following configuration:

```nix
{
  services.vaultwarden = {
    enable = true;
    environmentFile = config.sops.secrets."vaultwarden.env".path;
    dbBackend = "sqlite";
    config = {
      domain = "https://bitwarden.***";
      rocketPort = 41729;
      rocketWorkers = 20;
      signupsAllowed = false;
      websocketEnabled = true;
      websocketAddress = "0.0.0.0";
      websocketPort = 3012;
    };
  };

  # all recommended settings, ACME, etc. are enabled
  services.nginx.virtualHosts."bitwarden.***" =
    let cfg = config.services.vaultwarden.config; in
    {
      enableACME = true;
      forceSSL = true;
      locations."/" = {
        proxyPass = "http://127.0.0.1:${toString cfg.rocketPort}";
        proxyWebsockets = true;
      };
      locations."/notifications/hub" = {
        proxyPass = "http://127.0.0.1:${toString cfg.websocketPort}";
        proxyWebsockets = true;
      };
      locations."/notifications/hub/negotiate" = {
        proxyPass = "http://127.0.0.1:${toString cfg.rocketPort}";
        proxyWebsockets = true;
      };
    };
}
```

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.26.0
* Web-vault version: v2022.10.0
* Running within Docker: false (Base: Unknown)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.35.4
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "/var/lib/bitwarden_rs/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "/var/lib/bitwarden_rs",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "/***/***/************/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*********.****.**",
  "domain_origin": "*****://*********.****.**",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 5 * * * *",
  "emergency_request_timeout_schedule": "0 5 * * * *",
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "/var/lib/bitwarden_rs/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "/var/lib/bitwarden_rs/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "/var/lib/bitwarden_rs/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": null,
  "smtp_from": "*****@*********.****.**",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "****.*******.***",
  "smtp_password": "***",
  "smtp_port": 465,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "**********@****.****.**",
  "templates_folder": "/var/lib/bitwarden_rs/templates",
  "tmp_folder": "/var/lib/bitwarden_rs/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "/nix/store/7i89lb4kd8x32vql8miiwr2iklbvpyl5-vaultwarden-webvault-2022.10.0/share/vaultwarden/vault",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": "50964",
  "yubico_secret_key": "***",
  "yubico_server": null
}
```
</details>

**Comment by BlackDex** [comment]
You really need 1.27.0 to get the newer clients to work. Either upgrade Vaultwarden somehow, or downgrade the clients.

**Reply by xeals** [reply]
I didn't give exact versions, but all the Android clients I've used have not been the latest.  I've tested 2022.8.0 through 2022.10.0, where the latest (that requires 1.27.0) is 2022.11.0.

To my understanding it also shouldn't affect the browser versions as the version of the web UI that's bundled with 1.26.0 should be compatible with it, right?

**Reply by BlackDex** [reply]
The web-ui should work just fine indeed.
It probably has something to do with the connection between your reverse proxy and Vaultwarden. Check the ip on which Vaultwarden is running.

**Reply by xeals** [reply]
I've disabled the proxy entirely, connecting directly via IP and port, and seeing no effective change. Since nginx doesn't time out the request, the login just hangs.

**Comment by xeals** [comment, marked as answer]
It seems like my SMTP settings weren't properly migrated when the security variable was changed, and it might have been trying to connect with STARTTLS on the SSL/TLS port. When Vaultwarden tries to send the mail on login it appears to wait indefinitely on it and since it never establishes a connection, the login hangs. Forcing `SMTP_SECURITY` to `force_tls` resolves the issue.
